pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -263,17 +272,42 @@ func (r *reconciler) handleMirrorImage(ctx context.Context, targetImageRef strin
 	return nil
 }
 
-func checkAllBuildsSuccessful(builds map[string]*buildv1.Build) bool {
-	for _, build := range builds {
+func (r *reconciler) listBuilds(ctx context.Context, mabcName string) (*buildv1.BuildList, error) {
```

Comment:
I don't think that we should do that. I added the `status.builds` field in `mabc` for exactly this reason. Otherwise we will start spamming this call on every reconciliation. When the build is being created/updated, the controller updates the mabc and that's it. Also, listing those builds everytime beats the whole purpose of this PR.

Reply:
I meant this PR as removing the `build` reconciler and rely only on the `mabc` one.
There won't be any spamming for two reasons:
- we don't operate at huge scale, we can't optimize without having issues or numbers at hands
- most importantly, the controller makes heavy use of caching (see [pkg/cache/cache.go](https://github.com/kubernetes-sigs/controller-runtime/blob/3de96247226760a0b7c4c6c36e688d1cb41609b8/pkg/cache/cache.go)), the client doesn't always talk with the API server directly, especially for `GET` calls

This mechanism is well consolidated it won't overload the API Server unless we go really crazy in terms of build counts.

Reply:
The problem is that we will trigger the reconciliation for all builds running in the cluster. Removing the build reconciler, to my eyes, will make things worse. Now the build reconciler will be triggered only for the builds that were created by the controller. We update the mabc and that's it. I am not convinced why we should go ahead with this PR yet.

Reply:
> The problem is that we will trigger the reconciliation for all builds running in the cluster

No that's not true, see comment above [here](https://github.com/openshift/ci-tools/pull/3761#discussion_r1389606092)